jetty-websocket/websocket-javax-common/src/main/java/org/eclipse/jetty/websocket/javax/common/JavaxWebSocketBasicRemote.java

```diff
@@ -134,27 +134,32 @@ public void sendText(String partialMessage, boolean isLast) throws IOException
         {
             LOG.debug("sendText({},{})", TextUtil.hint(partialMessage), isLast);
         }
-        try (SharedBlockingCallback.Blocker b = session.getBlocking().acquire())
+
+        Frame frame;
+        switch (messageType)
         {
-            Frame frame;
-            switch (messageType)
-            {
-                case -1:
-                    // New message!
-                    frame = new Frame(OpCode.TEXT);
-                    break;
-                case OpCode.TEXT:
-                    frame = new Frame(OpCode.CONTINUATION);
-                    break;
-                case OpCode.BINARY:
-                    throw new IllegalStateException("Cannot send a partial TEXT message: BINARY message in progress");
-                default:
-                    throw new IllegalStateException("Cannot send a partial TEXT message: unrecognized active message type " + OpCode.name(messageType));
-            }
-
-            frame.setPayload(BufferUtil.toBuffer(partialMessage, UTF_8));
-            frame.setFin(isLast);
-            sendFrame(frame, b, false);
+            case -1:
+                // New message!
+                frame = new Frame(OpCode.TEXT);
+                break;
+            case OpCode.TEXT:
+                frame = new Frame(OpCode.CONTINUATION);
+                break;
+            case OpCode.BINARY:
+                throw new IllegalStateException("Cannot send a partial TEXT message: BINARY message in progress");
+            default:
+                throw new IllegalStateException("Cannot send a partial TEXT message: unrecognized active message type " + OpCode.name(messageType));
         }
+
+        frame.setPayload(BufferUtil.toBuffer(partialMessage, UTF_8));
+        frame.setFin(isLast);
+        BlockingCallback b = newBlockingCallback();
+        sendFrame(frame, b, false);
+        b.block();
+    }
+
+    private BlockingCallback newBlockingCallback()
+    {
+        return new BlockingCallback(getIdleTimeout() + 1000);
```

Comment:
if the idle timeout is 0, this is wrong.  You should only +1000 to timeouts >0

Reply:
oops yep I'll fix